app/controllers/api/v1/worksheets_controller.rb

```diff
@@ -18,7 +18,7 @@ def edit
 
   def update
     image_url = remote_image_url
-    prepped_params = worksheet_params.slice(:image_url, :photo).merge({ image_url: image_url })
+    prepped_params = worksheet_params.except(:image_url, :photo).merge({ image_url: image_url })
     if @worksheet.update(prepped_params)
```

Comment:
Then would this if statement run only when this method (in Class WorksheetPolicy):
 ```
 def index?
    case @user.role
    when 'admin'   then true
    when 'teacher' then @record.worksheet_template.user_id == @user.id
    when 'student' then @record.work_group.students.include?(@user)
    else false
    end
  end
```
returns true?

Reply:
This statement runs each time the controller is called (thanks to this `authorize` here: https://github.com/odaiba-app/server/pull/80/files#diff-e626411ac3c7c3e2593ab9895451461cR57 )
The statement returns `true` if the current user is allowed to see/modify the content, based on the conditions postulated in the code.